### PR TITLE
Add support for api-validator filter

### PR DESCRIPTION
**Pull request by vinod-m**
The current filter needs a wadl and supports rax roles.

### CODE REVIEW

File: recipes/filter-api-validator.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  owner node['repose']['owner']
+  group node['repose']['group']
+  mode '0644'
+  variables(
+    :wadl => node['repose']['api_validator']['wadl'],
+    :dot_output => node['repose']['api_validator']['dot_output'],
+    :enable_rax_roles => node['repose']['api_validator']['enable_rax_roles'],
+  )
+  notifies :restart, 'service[repose-valve]'
+end
```
Comment (filler) [inline recipes/filter-api-validator.rb]: Probably want to add a minitest test for this template resource.  There are examples in https://github.com/rackerlabs/cookbook-repose/tree/master/files/default/test

File: files/default/test/filter-api-validator.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+  it 'sets up validator.cfg.xml' do
+    file('/etc/repose/validator.cfg.xml').must_exist.with(
+      :owner, node['repose']['owner']).and(
+      :group, node['repose']['group']).and(
+      :mode, '0644')
+  end
+
+end
+
```
Comment (filler) [inline files/default/test/filter-api-validator.rb]: :metal:

**Comment by filler** [comment]
:100: for RBAC!  Overall, I think this is great.

In order to people to use the filter, they need to leverage their own wrapper cookbook to satisfy resources for `node['repose']['api_validator']['wadl']` and `node['repose']['api_validator']['dot_output']`.  Can you clarify that in the `README` @vinod-m?  

Thanks!

**Comment by filler** [comment]
@b0d0nne11 I cant think of a way to satisfy the base `wadl` and `dot` from a usability perspective.  I think users have to bring their own with a wrapper.  Or is there some sort of no-op that Im not thinking of?

**Comment by b0d0nne11** [comment]
At a minimum we should probably set explicit, empty defaults for them and then make it clear in the documentation that they need to be overridden to use the filter. Users would also be responsible for creating the files either in a wrapper or in their application cookbook and that should also be clear.

The other option is to supply a no-op validator (I'm not sure what that looks like) and then allow users to override the `cookbook` attribute of any `template` or `cookbook_file` resources but that a lot more complicated without much gain.

**Comment by filler** [comment]
I think thats fine.  `nil` attribs for the `wadl` and `dot`.  Also some words in the `README` along the lines of
> If you want to use this, you need to bring your own wrapper cookbook and satisfy the `wadl`

**Comment by vinod-m** [comment]
I updated the readme file, bumped the version and put in nil attribs for the wadl and dot-output file.

**Comment by b0d0nne11** [comment]
LGTM, what do you think @filler ?

**Comment by filler** [comment]
:+1: lgtm

**Comment by filler** [comment]
@b0d0nne11 can you merge?  Looks like I lost authz. :sob: 

> This pull request can be automatically merged by project collaborators.
> Only those with write access to this repository can merge pull requests.

**Comment by b0d0nne11** [comment]
@filler Nope, same here. :(
@dimtruck Do you know who has access or who we should ask to get re-added?

**Comment by dimtruck** [comment]
Looks like it's just team-repose that has access (repose devs + qe).  Maybe we need a different group?

**Comment by filler** [comment]
At first I was all 
 

... then I was all 


Thanks, @dimtruck.  Whatever you can do to re-add @b0d0nne11 and I would be boss.  Thanks.

**Pull request merged**